module/zfs/arc.c

```diff
@@ -9729,31 +9936,38 @@ l2arc_remove_vdev(vdev_t *vd)
 		remdev->l2ad_rebuild_cancel = B_TRUE;
 		while (remdev->l2ad_rebuild == B_TRUE)
 			cv_wait(&l2arc_rebuild_thr_cv, &l2arc_rebuild_thr_lock);
+	} else if (remdev->l2ad_rebuild == B_TRUE) {
+		/* Rebuild hasn't started yet so skip asynchronous teardown */
+		asynchronous = B_FALSE;
```

Comment:
I wonder why do we need to go to synchronous eviction if rebuild haven't started yet? I think we could just move `remdev->l2ad_rebuild_cancel = B_TRUE;` before the `if (remdev->l2ad_rebuild_began == B_TRUE) {` and rebuild should just never start.

Reply:
Yep, good point.  If we know that it's being removed then we could stop the rebuild.  I was initially concerned with racing here